unison-runtime/src/Unison/Runtime/MCode.hs

```diff
@@ -722,6 +792,38 @@ emitCombs rns grpr grpn (Rec grp ent) =
     rec = M.fromList $ zip rvs ixs
     aux = foldMap (emitComb rns grpr grpn rec) (zip ixs cmbs)
 
+-- | lazily replace all references to combinators with the combinators themselves,
+-- tying the knot recursively when necessary.
+resolveCombs ::
+  -- Existing in-scope combs that might be referenced
+  -- TODO: Do we ever actually need to pass this?
+  Maybe (EnumMap Word64 RCombs) ->
+  -- Combinators which need their knots tied.
+  EnumMap Word64 Combs ->
+  EnumMap Word64 RCombs
+resolveCombs mayExisting combs =
+  -- Fixed point lookup;
+  -- We make sure not to force resolved Combs or we'll loop forever.
+  let ~resolved =
```

Comment:
This `~` doesn't actually do anything. `let` is irrefutable by default unless perhaps the `Strict` extension is enabled, which I wouldn't recommend, because what it actually does can be rather confusing.

Reply:
I can remove it if you like, I'm mostly using it as a sign-post since  Haskell doesn't have explicit let-recs 😢